src/dfx/assets/language_bindings/index.js.hbs

```diff
@@ -1,25 +1,43 @@
 import { Actor, HttpAgent } from "@dfinity/agent";
 
 // Imports and re-exports candid interface
-import { idlFactory } from './{{canister_name}}.did.js';
-export { idlFactory } from './{{canister_name}}.did.js';
+import { idlFactory } from "./{{canister_name}}.did.js";
+export { idlFactory } from "./{{canister_name}}.did.js";
 // CANISTER_ID is replaced by webpack based on node environment
-export const canisterId = {{canister_name_process_env}};
+export const canisterId = {{{canister_name_process_env}}};
 
 {{{{raw}}}}
 /**
- * 
+ * @typedef CreateActorOptions
+ * @property {(import("@dfinity/agent").Agent)} [agent]
+ * @property {(import("@dfinity/agent").HttpAgentOptions)} [agentOptions]
+ * @property {(import("@dfinity/agent").ActorConfig)} [actorOptions]
+ */
+
+/**
+ *
  * @param {string | import("@dfinity/principal").Principal} canisterId Canister ID of Agent
- * @param {{agentOptions?: import(\"@dfinity/agent\").HttpAgentOptions; actorOptions?: import(\"@dfinity/agent\").ActorConfig}} [options]
- * @return {import("@dfinity/agent").ActorSubclass<import("{{{{/raw}}}}./{{canister_name}}.did.js")._SERVICE>}
+ * @param {CreateActorOptions} options {@link CreateActorOptions}
+ * @param {CreateActorOptions["agent"]} [options.agent] An initialized agent
+ * @param {CreateActorOptions["agentOptions"]} [options.agentOptions] Options to initialize an {@link HttpAgent}. Overridden if an `agent` is passed.
+ * @param {CreateActorOptions["actorOptions"]} [options.actorOptions] Options of to pass during the actor initialization.
+ * @return {import("@dfinity/agent").ActorSubclass<import("{{{{/raw}}}}./{{canister_name}}.did.js")._SERVICE>} ActorSubclass configured for the canister
  */
-export const createActor = (canisterId, options) => {
-  const agent = new HttpAgent(options ? { ...options.agentOptions } : {});
-  
+export const createActor = (canisterId, options = {}) => {
+  const agent = options.agent || new HttpAgent({ ...options.agentOptions });
+
+  if (options.agent && options.agentConfig) {
```

Comment:
shouldn't this be checking for options.agentOptions? 
```suggestion
  if (options.agent && options.agentOptions) {
```

Reply:
Yes, thanks for catching